automation_tools/satellite6/upgrade/capsule.py

```diff
@@ -0,0 +1,161 @@
+    perform capsule upgrade later
+
+    :param string sat_host: Satellite hostname to which the capsule registered
+    :param string os_version: The OS version onto which the capsule installed
+        e.g: rhel6, rhel7
+    """
+    # If Personal Capsule Hostname provided
+    if os.environ.get('CAPSULE_HOSTNAMES'):
+        cap_hosts = os.environ.get('CAPSULE_HOSTNAMES')
+        # For Personal Satellite CAPSULE_SUBSCRIPTION is must
+        cap_subscription = os.environ.get('CAPSULE_SUBSCRIPTION')
+        if not cap_subscription:
+            print('CAPSULE_SUBSCRIPTION environment variable is not '
+                  'defined !')
+            sys.exit(1)
+        elif len(cap_subscription.split(',')) != 3:
+            print('CAPSULE_SUBSCRIPTION environment variable is not '
+                  'having all the details!')
+    # Else run upgrade on rhevm capsule
+    else:
+        # Get image name and Hostname from Jenkins environment
+        missing_vars = [
+            var for var in (
+                'RHEV_CAP_IMAGE',
+                'RHEV_CAP_HOST',
+                'RHEV_CAPSULE_CV',
+                'RHEV_CAPSULE_ENVIRONMENT',
+                'RHEV_CAPSULE_AK')
+            if var not in os.environ]
+        # Check if image name and Hostname in jenkins are set
+        if missing_vars:
+            print('The following jenkins environment variable(s) must be '
+                  'set: {0}.'.format(', '.join(missing_vars)))
+            sys.exit(1)
+        cap_image = os.environ.get('RHEV_CAP_IMAGE')
+        cap_hosts = os.environ.get('RHEV_CAP_HOST')
+        cap_instance = 'upgrade_capsule_auto_{0}'.format(os_version)
+        # Deleting Capsule instance if already exists
+        execute(delete_rhevm_instance, cap_instance)
+        print('Turning on Capsule Instance ....')
+        execute(create_rhevm_instance, cap_instance, cap_image)
+        # Restarting the services on capsule
+        execute(lambda: run('katello-service restart'), host=cap_hosts)
+    # Set capsule hostname in fabric environment
+    env['capsule_hosts'] = cap_hosts
+    # Copy ssh key from satellie to capsule
+    copy_ssh_key(sat_host, cap_hosts)
+    if os.environ.get('CAPSULE_URL') is not None:
+        execute(sync_capsule_repos_to_upgrade, cap_hosts, host=sat_host)
+
+
+def satellite6_capsule_upgrade(cap_host):
+    """Upgrades capsule from existing version to latest version.
+
+    :param string cap_host: Capsule hostname onto which the capsule upgrade
+    will run
+
+    The following environment variables affect this command:
+
+    CAPSULE_URL
+        Optional, defaults to available capsule version in CDN.
+        URL for capsule of latest compose to upgrade.
+    FROM_VERSION
+        Capsule current version, to disable repos while upgrading.
+        e.g '6.1','6.0'
+    TO_VERSION
+        Capsule version to upgrade to and enable repos while upgrading.
+        e.g '6.1','6.2'
+
+    """
+    sat_host = env.get('satellite_host')
```

Comment:
You're right, got confused with env vars. This is actually a Fabric's env variable not system env var.